From 59ef53ade1c80278c1c17c2fad38ed5ef15749c7 Mon Sep 17 00:00:00 2001
From: Igor Plesser <i.plesser@inango-systems.com>
Date: Wed, 21 May 2025 21:36:01 +0300
Subject: [PATCH] boardfarm: add crash check support

- Add helper for ba-cli
- Introduce @process_fault_check decorator for ProcessFaults validation
- Integrate crash check into all existing tests

PPM-3241

Signed-off-by: Igor Plesser <i.plesser@inango-systems.com>
---
 .../tests/ap_capability_query.py              |  2 +
 .../tests/ap_config_bss_tear_down.py          |  2 +
 .../tests/ap_config_bss_tear_down_cli.py      |  5 ++-
 .../tests/ap_config_renew.py                  |  7 ++--
 .../tests/ap_metrics_response.py              |  2 +
 .../tests/ap_operational_bss_tlv.py           |  2 +
 .../tests/beacon_report_query.py              |  2 +
 .../tests/beacon_report_query_and_response.py |  2 +
 .../boardfarm_prplmesh/tests/cac_request.py   |  2 +
 .../tests/capi_wireless_onboarding.py         |  2 +
 .../tests/channel_selection.py                |  7 ++--
 .../tests/client_association.py               |  2 +
 .../tests/client_association_dummy.py         |  2 +
 .../tests/client_association_link_metrics.py  |  2 +
 ...ient_capability_query_fails_with_no_sta.py |  2 +
 .../client_capability_query_successful.py     |  2 +
 .../tests/client_steering_dummy.py            |  1 +
 .../tests/client_steering_mandate.py          |  1 +
 .../tests/combined_infra_metrics.py           |  2 +
 .../tests/dev_reset_default.py                |  2 +
 .../higher_layer_data_payload_trigger.py      |  2 +
 .../tests/initial_ap_config.py                |  2 +
 .../tests/iperf3_test_run.py                  |  2 +
 .../tests/link_metric_query.py                |  2 +
 .../boardfarm_prplmesh/tests/link_metrics.py  |  2 +
 .../tests/maximum_throughput_tcp.py           |  2 +
 ...policy_config_w_metric_reporting_policy.py |  2 +
 ...ulti_ap_policy_config_w_steering_policy.py |  3 +-
 ...policy_config_w_unsucessful_association.py |  3 +-
 .../tests/nbapi_access_point.py               |  3 +-
 .../tests/nbapi_association_event.py          |  3 +-
 .../boardfarm_prplmesh/tests/nbapi_bss.py     |  2 +
 .../boardfarm_prplmesh/tests/nbapi_cac.py     |  3 +-
 .../tests/nbapi_capabilities.py               |  2 +
 .../boardfarm_prplmesh/tests/nbapi_device.py  |  3 +-
 .../tests/nbapi_disassociation_event.py       |  3 +-
 .../boardfarm_prplmesh/tests/nbapi_radio.py   |  2 +
 .../tests/nbapi_radio_backhaul_sta.py         |  3 +-
 .../tests/nbapi_scan_result.py                |  3 +-
 .../boardfarm_prplmesh/tests/nbapi_sta.py     |  3 +-
 .../tests/nbapi_triger_scan.py                |  3 +-
 .../tests/remote_station.py                   |  3 +-
 .../tests/resources_test_run.py               |  3 +-
 .../tests/simulate_v2_certification_4_7_10.py |  3 +-
 .../boardfarm_prplmesh/tests/topology.py      |  3 +-
 .../tests/topology_discovery_burst.py         |  3 +-
 .../tests/tunnelled_frames.py                 |  3 +-
 tests/environment.py                          | 39 +++++++++++++++++++
 48 files changed, 136 insertions(+), 25 deletions(-)

diff --git a/tests/boardfarm_plugins/boardfarm_prplmesh/tests/ap_capability_query.py b/tests/boardfarm_plugins/boardfarm_prplmesh/tests/ap_capability_query.py
index f6c54a1d6f..56e5c49740 100644
--- a/tests/boardfarm_plugins/boardfarm_prplmesh/tests/ap_capability_query.py
+++ b/tests/boardfarm_plugins/boardfarm_prplmesh/tests/ap_capability_query.py
@@ -7,10 +7,12 @@ from .prplmesh_base_test import PrplMeshBaseTest
 from boardfarm.exceptions import SkipTest
 from opts import debug
 import time
+import environment as env
 
 
 class ApCapabilityQuery(PrplMeshBaseTest):
 
+    @env.process_faults_check
     def runTest(self):
         # Locate test participants
         try:
diff --git a/tests/boardfarm_plugins/boardfarm_prplmesh/tests/ap_config_bss_tear_down.py b/tests/boardfarm_plugins/boardfarm_prplmesh/tests/ap_config_bss_tear_down.py
index ed3e196654..b484e98196 100644
--- a/tests/boardfarm_plugins/boardfarm_prplmesh/tests/ap_config_bss_tear_down.py
+++ b/tests/boardfarm_plugins/boardfarm_prplmesh/tests/ap_config_bss_tear_down.py
@@ -6,6 +6,7 @@
 from .prplmesh_base_test import PrplMeshBaseTest
 from boardfarm.exceptions import SkipTest
 from capi import tlv
+import environment as env
 
 
 class ApConfigBSSTeardown(PrplMeshBaseTest):
@@ -17,6 +18,7 @@ class ApConfigBSSTeardown(PrplMeshBaseTest):
 
     """
 
+    @env.process_faults_check
     def runTest(self):
         # Locate test participants
         try:
diff --git a/tests/boardfarm_plugins/boardfarm_prplmesh/tests/ap_config_bss_tear_down_cli.py b/tests/boardfarm_plugins/boardfarm_prplmesh/tests/ap_config_bss_tear_down_cli.py
index 047c372615..5421f51228 100644
--- a/tests/boardfarm_plugins/boardfarm_prplmesh/tests/ap_config_bss_tear_down_cli.py
+++ b/tests/boardfarm_plugins/boardfarm_prplmesh/tests/ap_config_bss_tear_down_cli.py
@@ -5,7 +5,7 @@
 
 from .prplmesh_base_test import PrplMeshBaseTest
 from boardfarm.exceptions import SkipTest
-import environment
+import environment as env
 
 
 class ApConfigBSSTeardownCli(PrplMeshBaseTest):
@@ -17,6 +17,7 @@ class ApConfigBSSTeardownCli(PrplMeshBaseTest):
 
     """
 
+    @env.process_faults_check
     def runTest(self):
         # Locate test participants
         try:
@@ -51,7 +52,7 @@ class ApConfigBSSTeardownCli(PrplMeshBaseTest):
 
         vap_bss_type = radio_0_vap_0.get_bss_type()
 
-        if vap_bss_type != environment.BssType.Fronthaul:
+        if vap_bss_type != env.BssType.Fronthaul:
             self.fail(
                 f"Radio 0 vap {ssid} bss type is {vap_bss_type.name} "
                 "when it should be Fronthaul")
diff --git a/tests/boardfarm_plugins/boardfarm_prplmesh/tests/ap_config_renew.py b/tests/boardfarm_plugins/boardfarm_prplmesh/tests/ap_config_renew.py
index c02420463d..31807cb4cf 100644
--- a/tests/boardfarm_plugins/boardfarm_prplmesh/tests/ap_config_renew.py
+++ b/tests/boardfarm_plugins/boardfarm_prplmesh/tests/ap_config_renew.py
@@ -8,12 +8,13 @@ import time
 
 from .prplmesh_base_test import PrplMeshBaseTest
 from capi import tlv
-import environment
+import environment as env
 
 
 class ApConfigRenew(PrplMeshBaseTest):
     """Check initial configuration on device."""
 
+    @env.process_faults_check
     def runTest(self):
         # Locate test participants
         agent = self.dev.DUT.agent_entity
@@ -52,7 +53,7 @@ class ApConfigRenew(PrplMeshBaseTest):
 
         vap_bss_type = radio_0_vap_0.get_bss_type()
 
-        if vap_bss_type != environment.BssType.Fronthaul:
+        if vap_bss_type != env.BssType.Fronthaul:
             self.fail(
                 f"Radio 0 vap {ssid_1} bss type is {vap_bss_type.name}"
                 " when it should be Fronthaul")
@@ -64,7 +65,7 @@ class ApConfigRenew(PrplMeshBaseTest):
 
         vap_bss_type = radio_0_vap_1.get_bss_type()
 
-        if vap_bss_type != environment.BssType.Backhaul:
+        if vap_bss_type != env.BssType.Backhaul:
             self.fail(
                 f"Radio 1 vap {ssid_2} bss type is {vap_bss_type.name}"
                 " when it should be Backhaul")
diff --git a/tests/boardfarm_plugins/boardfarm_prplmesh/tests/ap_metrics_response.py b/tests/boardfarm_plugins/boardfarm_prplmesh/tests/ap_metrics_response.py
index 0e46f36043..0410df5bd5 100644
--- a/tests/boardfarm_plugins/boardfarm_prplmesh/tests/ap_metrics_response.py
+++ b/tests/boardfarm_plugins/boardfarm_prplmesh/tests/ap_metrics_response.py
@@ -11,6 +11,7 @@ from capi import tlv
 from opts import debug
 
 import time
+import environment as env
 
 
 class ApMetricsResponse(PrplMeshBaseTest):
@@ -26,6 +27,7 @@ class ApMetricsResponse(PrplMeshBaseTest):
         GW - Controller
     '''
 
+    @env.process_faults_check
     def runTest(self):
         # Locate test participants
         try:
diff --git a/tests/boardfarm_plugins/boardfarm_prplmesh/tests/ap_operational_bss_tlv.py b/tests/boardfarm_plugins/boardfarm_prplmesh/tests/ap_operational_bss_tlv.py
index de3479430f..6d331748eb 100644
--- a/tests/boardfarm_plugins/boardfarm_prplmesh/tests/ap_operational_bss_tlv.py
+++ b/tests/boardfarm_plugins/boardfarm_prplmesh/tests/ap_operational_bss_tlv.py
@@ -10,6 +10,7 @@ from boardfarm.exceptions import SkipTest
 from opts import debug
 
 import time
+import environment as env
 
 
 class ApOperationalBss(PrplMeshBaseTest):
@@ -17,6 +18,7 @@ class ApOperationalBss(PrplMeshBaseTest):
         Additionally, check the currently used BSS type
     '''
 
+    @env.process_faults_check
     def runTest(self):
         try:
             controller = self.dev.lan.controller_entity
diff --git a/tests/boardfarm_plugins/boardfarm_prplmesh/tests/beacon_report_query.py b/tests/boardfarm_plugins/boardfarm_prplmesh/tests/beacon_report_query.py
index f07716f649..0cb593d4aa 100644
--- a/tests/boardfarm_plugins/boardfarm_prplmesh/tests/beacon_report_query.py
+++ b/tests/boardfarm_plugins/boardfarm_prplmesh/tests/beacon_report_query.py
@@ -8,10 +8,12 @@ from boardfarm.exceptions import SkipTest
 from capi import tlv
 from opts import debug
 import time
+import environment as env
 
 
 class BeaconReportQuery(PrplMeshBaseTest):
 
+    @env.process_faults_check
     def runTest(self):
         # Locate test participants
         try:
diff --git a/tests/boardfarm_plugins/boardfarm_prplmesh/tests/beacon_report_query_and_response.py b/tests/boardfarm_plugins/boardfarm_prplmesh/tests/beacon_report_query_and_response.py
index 6801654167..bb88ccc76c 100644
--- a/tests/boardfarm_plugins/boardfarm_prplmesh/tests/beacon_report_query_and_response.py
+++ b/tests/boardfarm_plugins/boardfarm_prplmesh/tests/beacon_report_query_and_response.py
@@ -11,6 +11,7 @@ from capi import tlv
 from opts import debug
 
 import time
+import environment as env
 
 
 class BeaconReportQueryAndResponse(PrplMeshBaseTest):
@@ -19,6 +20,7 @@ class BeaconReportQueryAndResponse(PrplMeshBaseTest):
     receiving a response from the STA, and sending the contents of that response
     in a Beacon Metrics Response message to the Controller '''
 
+    @env.process_faults_check
     def runTest(self):
         # Locate test participants
         try:
diff --git a/tests/boardfarm_plugins/boardfarm_prplmesh/tests/cac_request.py b/tests/boardfarm_plugins/boardfarm_prplmesh/tests/cac_request.py
index f920a1f7d8..4529f3738d 100644
--- a/tests/boardfarm_plugins/boardfarm_prplmesh/tests/cac_request.py
+++ b/tests/boardfarm_plugins/boardfarm_prplmesh/tests/cac_request.py
@@ -8,10 +8,12 @@ from boardfarm.exceptions import SkipTest
 from capi import tlv
 #  from opts import debug
 import time
+import environment as env
 
 
 class CacRequest(PrplMeshBaseTest):
 
+    @env.process_faults_check
     def runTest(self):
         # Locate test participants
         try:
diff --git a/tests/boardfarm_plugins/boardfarm_prplmesh/tests/capi_wireless_onboarding.py b/tests/boardfarm_plugins/boardfarm_prplmesh/tests/capi_wireless_onboarding.py
index 473ee8c87f..9ea39e361c 100644
--- a/tests/boardfarm_plugins/boardfarm_prplmesh/tests/capi_wireless_onboarding.py
+++ b/tests/boardfarm_plugins/boardfarm_prplmesh/tests/capi_wireless_onboarding.py
@@ -8,6 +8,7 @@ import time
 
 from .prplmesh_base_test import PrplMeshBaseTest
 from boardfarm.exceptions import SkipTest
+import environment as env
 
 
 class CapiWirelessOnboarding(PrplMeshBaseTest):
@@ -18,6 +19,7 @@ class CapiWirelessOnboarding(PrplMeshBaseTest):
         Checks if prplMesh uses the AP1 radio 0 MAC as backhaul MAC after the onboarding process
     """
 
+    @env.process_faults_check
     def runTest(self):
         # Locate test participants
         try:
diff --git a/tests/boardfarm_plugins/boardfarm_prplmesh/tests/channel_selection.py b/tests/boardfarm_plugins/boardfarm_prplmesh/tests/channel_selection.py
index f575fe7cd4..86f343c210 100644
--- a/tests/boardfarm_plugins/boardfarm_prplmesh/tests/channel_selection.py
+++ b/tests/boardfarm_plugins/boardfarm_prplmesh/tests/channel_selection.py
@@ -9,8 +9,8 @@ import time
 from .prplmesh_base_test import PrplMeshBaseTest
 from boardfarm.exceptions import SkipTest
 from capi import tlv
-from environment import ChannelTlvs
 from opts import debug
+import environment as env
 
 
 class ChannelSelection(PrplMeshBaseTest):
@@ -30,6 +30,7 @@ class ChannelSelection(PrplMeshBaseTest):
     - Should fail if channels haven't changed to 6 and 36
     """
 
+    @env.process_faults_check
     def runTest(self):
 
         def check_single_channel_response(self, resp_code) -> None:
@@ -187,10 +188,10 @@ class ChannelSelection(PrplMeshBaseTest):
         self.checkpoint()
 
         # payload_wlan0 - request for change channel on 6
-        payload_wlan0 = ChannelTlvs.CHANNEL_6.value
+        payload_wlan0 = env.ChannelTlvs.CHANNEL_6.value
 
         # payload_wlan2  - request for change channel on 36
-        payload_wlan2 = ChannelTlvs.CHANNEL_36.value
+        payload_wlan2 = env.ChannelTlvs.CHANNEL_36.value
 
         """
         Step 1: Trigger channel selection to channel 6 and 36. Check that
diff --git a/tests/boardfarm_plugins/boardfarm_prplmesh/tests/client_association.py b/tests/boardfarm_plugins/boardfarm_prplmesh/tests/client_association.py
index 73e2f4fd83..ed0727c4f3 100644
--- a/tests/boardfarm_plugins/boardfarm_prplmesh/tests/client_association.py
+++ b/tests/boardfarm_plugins/boardfarm_prplmesh/tests/client_association.py
@@ -7,6 +7,7 @@ from .prplmesh_base_test import PrplMeshBaseTest
 from boardfarm.exceptions import SkipTest
 from capi import tlv
 from opts import debug
+import environment as env
 
 
 class ClientAssociation(PrplMeshBaseTest):
@@ -23,6 +24,7 @@ class ClientAssociation(PrplMeshBaseTest):
         GW controller if checked for a ACK message
     """
 
+    @env.process_faults_check
     def runTest(self):
         # Locate test participants
         try:
diff --git a/tests/boardfarm_plugins/boardfarm_prplmesh/tests/client_association_dummy.py b/tests/boardfarm_plugins/boardfarm_prplmesh/tests/client_association_dummy.py
index 717c868a3f..8430b21383 100644
--- a/tests/boardfarm_plugins/boardfarm_prplmesh/tests/client_association_dummy.py
+++ b/tests/boardfarm_plugins/boardfarm_prplmesh/tests/client_association_dummy.py
@@ -9,6 +9,7 @@ import time
 from .prplmesh_base_test import PrplMeshBaseTest
 from boardfarm.exceptions import SkipTest
 from opts import debug
+import environment as env
 
 
 class ClientAssociationDummy(PrplMeshBaseTest):
@@ -33,6 +34,7 @@ class ClientAssociationDummy(PrplMeshBaseTest):
         Connection map is checked for repeater 1 wlan0
     """
 
+    @env.process_faults_check
     def runTest(self):
         # Locate test participants
         try:
diff --git a/tests/boardfarm_plugins/boardfarm_prplmesh/tests/client_association_link_metrics.py b/tests/boardfarm_plugins/boardfarm_prplmesh/tests/client_association_link_metrics.py
index e183dee5e3..95b572ef90 100644
--- a/tests/boardfarm_plugins/boardfarm_prplmesh/tests/client_association_link_metrics.py
+++ b/tests/boardfarm_plugins/boardfarm_prplmesh/tests/client_association_link_metrics.py
@@ -10,6 +10,7 @@ import time
 from .prplmesh_base_test import PrplMeshBaseTest
 from capi import tlv
 from opts import debug
+import environment as env
 
 
 class ClientAssociationLinkMetrics(PrplMeshBaseTest):
@@ -17,6 +18,7 @@ class ClientAssociationLinkMetrics(PrplMeshBaseTest):
     an Associated STA Link Metrics Query message with an Associated STA Link Metrics
     Response message containing an Associated STA Link Metrics TLV for the associated STA.'''
 
+    @env.process_faults_check
     def runTest(self):
         # Locate test participants
         agent = self.dev.DUT.agent_entity
diff --git a/tests/boardfarm_plugins/boardfarm_prplmesh/tests/client_capability_query_fails_with_no_sta.py b/tests/boardfarm_plugins/boardfarm_prplmesh/tests/client_capability_query_fails_with_no_sta.py
index 5060fdac30..77f13aa682 100644
--- a/tests/boardfarm_plugins/boardfarm_prplmesh/tests/client_capability_query_fails_with_no_sta.py
+++ b/tests/boardfarm_plugins/boardfarm_prplmesh/tests/client_capability_query_fails_with_no_sta.py
@@ -8,6 +8,7 @@
 from .prplmesh_base_test import PrplMeshBaseTest
 from boardfarm.exceptions import SkipTest
 from opts import debug
+import environment as env
 
 
 class ClientCapabilityQueryFailsWithNoSta(PrplMeshBaseTest):
@@ -22,6 +23,7 @@ class ClientCapabilityQueryFailsWithNoSta(PrplMeshBaseTest):
         The action should return an error message
     """
 
+    @env.process_faults_check
     def runTest(self):
         # Locate test participants
         try:
diff --git a/tests/boardfarm_plugins/boardfarm_prplmesh/tests/client_capability_query_successful.py b/tests/boardfarm_plugins/boardfarm_prplmesh/tests/client_capability_query_successful.py
index 42df2dcd86..1c39f81c58 100644
--- a/tests/boardfarm_plugins/boardfarm_prplmesh/tests/client_capability_query_successful.py
+++ b/tests/boardfarm_plugins/boardfarm_prplmesh/tests/client_capability_query_successful.py
@@ -8,6 +8,7 @@ from .prplmesh_base_test import PrplMeshBaseTest
 from boardfarm.exceptions import SkipTest
 from opts import debug
 import time
+import environment as env
 
 
 class ClientCapabilityQuerySuccessful(PrplMeshBaseTest):
@@ -22,6 +23,7 @@ class ClientCapabilityQuerySuccessful(PrplMeshBaseTest):
         The there should be an expected capability frame on the report
     """
 
+    @env.process_faults_check
     def runTest(self):
         # Locate test participants
         try:
diff --git a/tests/boardfarm_plugins/boardfarm_prplmesh/tests/client_steering_dummy.py b/tests/boardfarm_plugins/boardfarm_prplmesh/tests/client_steering_dummy.py
index 6342e7f59d..ae67688f76 100644
--- a/tests/boardfarm_plugins/boardfarm_prplmesh/tests/client_steering_dummy.py
+++ b/tests/boardfarm_plugins/boardfarm_prplmesh/tests/client_steering_dummy.py
@@ -37,6 +37,7 @@ class ClientSteeringDummy(PrplMeshBaseTest):
         Value of BTMSuccesses should be incremented by one
     """
 
+    @env.process_faults_check
     def runTest(self):
         # Locate test participants
         try:
diff --git a/tests/boardfarm_plugins/boardfarm_prplmesh/tests/client_steering_mandate.py b/tests/boardfarm_plugins/boardfarm_prplmesh/tests/client_steering_mandate.py
index 452a466aef..949b2c8bae 100644
--- a/tests/boardfarm_plugins/boardfarm_prplmesh/tests/client_steering_mandate.py
+++ b/tests/boardfarm_plugins/boardfarm_prplmesh/tests/client_steering_mandate.py
@@ -20,6 +20,7 @@ class ClientSteeringMandate(PrplMeshBaseTest):
         GW - Controller
     """
 
+    @env.process_faults_check
     def runTest(self):
         # Locate test participants
         try:
diff --git a/tests/boardfarm_plugins/boardfarm_prplmesh/tests/combined_infra_metrics.py b/tests/boardfarm_plugins/boardfarm_prplmesh/tests/combined_infra_metrics.py
index a931500dcf..c9a496b699 100644
--- a/tests/boardfarm_plugins/boardfarm_prplmesh/tests/combined_infra_metrics.py
+++ b/tests/boardfarm_plugins/boardfarm_prplmesh/tests/combined_infra_metrics.py
@@ -11,6 +11,7 @@ from capi import tlv
 from opts import debug
 
 import time
+import environment as env
 
 
 class CombinedInfraMetrics(PrplMeshBaseTest):
@@ -24,6 +25,7 @@ class CombinedInfraMetrics(PrplMeshBaseTest):
         GW - Controller
     """
 
+    @env.process_faults_check
     def runTest(self):
         # Locate test participants
         try:
diff --git a/tests/boardfarm_plugins/boardfarm_prplmesh/tests/dev_reset_default.py b/tests/boardfarm_plugins/boardfarm_prplmesh/tests/dev_reset_default.py
index 0fea113e97..b75eb60c3f 100644
--- a/tests/boardfarm_plugins/boardfarm_prplmesh/tests/dev_reset_default.py
+++ b/tests/boardfarm_plugins/boardfarm_prplmesh/tests/dev_reset_default.py
@@ -8,6 +8,7 @@ import time
 
 from .prplmesh_base_test import PrplMeshBaseTest
 from boardfarm.exceptions import SkipTest
+import environment as env
 
 
 class DevResetDefault(PrplMeshBaseTest):
@@ -20,6 +21,7 @@ class DevResetDefault(PrplMeshBaseTest):
 
     """
 
+    @env.process_faults_check
     def runTest(self):
         # Locate test participants
         try:
diff --git a/tests/boardfarm_plugins/boardfarm_prplmesh/tests/higher_layer_data_payload_trigger.py b/tests/boardfarm_plugins/boardfarm_prplmesh/tests/higher_layer_data_payload_trigger.py
index 115064c866..d26fa5c127 100644
--- a/tests/boardfarm_plugins/boardfarm_prplmesh/tests/higher_layer_data_payload_trigger.py
+++ b/tests/boardfarm_plugins/boardfarm_prplmesh/tests/higher_layer_data_payload_trigger.py
@@ -8,10 +8,12 @@ from boardfarm.exceptions import SkipTest
 from capi import tlv
 from opts import debug
 import time
+import environment as env
 
 
 class HigherLayerDataPayloadTrigger(PrplMeshBaseTest):
 
+    @env.process_faults_check
     def runTest(self):
         # Locate test participants
         try:
diff --git a/tests/boardfarm_plugins/boardfarm_prplmesh/tests/initial_ap_config.py b/tests/boardfarm_plugins/boardfarm_prplmesh/tests/initial_ap_config.py
index 56a08dc3da..69faadf000 100644
--- a/tests/boardfarm_plugins/boardfarm_prplmesh/tests/initial_ap_config.py
+++ b/tests/boardfarm_plugins/boardfarm_prplmesh/tests/initial_ap_config.py
@@ -4,11 +4,13 @@
 # See LICENSE file for more details.
 
 from .prplmesh_base_test import PrplMeshBaseTest
+import environment as env
 
 
 class InitialApConfig(PrplMeshBaseTest):
     """Check initial configuration on device."""
 
+    @env.process_faults_check
     def runTest(self):
         # Locate test participants
         agent = self.dev.DUT.agent_entity
diff --git a/tests/boardfarm_plugins/boardfarm_prplmesh/tests/iperf3_test_run.py b/tests/boardfarm_plugins/boardfarm_prplmesh/tests/iperf3_test_run.py
index 2ae6bbf649..35211ec113 100644
--- a/tests/boardfarm_plugins/boardfarm_prplmesh/tests/iperf3_test_run.py
+++ b/tests/boardfarm_plugins/boardfarm_prplmesh/tests/iperf3_test_run.py
@@ -1,9 +1,11 @@
 from .prplmesh_base_test import PrplMeshBaseTest
+import environment as env
 
 
 class Iperf3TestRun(PrplMeshBaseTest):
     """Check initial configuration on device."""
 
+    @env.process_faults_check
     def runTest(self):
 
         agent = self.dev.DUT.agent_entity
diff --git a/tests/boardfarm_plugins/boardfarm_prplmesh/tests/link_metric_query.py b/tests/boardfarm_plugins/boardfarm_prplmesh/tests/link_metric_query.py
index d579a3e4f7..467420dc56 100644
--- a/tests/boardfarm_plugins/boardfarm_prplmesh/tests/link_metric_query.py
+++ b/tests/boardfarm_plugins/boardfarm_prplmesh/tests/link_metric_query.py
@@ -4,6 +4,7 @@ from boardfarm.exceptions import SkipTest
 from .prplmesh_base_test import PrplMeshBaseTest
 from capi import tlv
 from opts import debug
+import environment as env
 
 
 class LinkMetricQuery(PrplMeshBaseTest):
@@ -35,6 +36,7 @@ class LinkMetricQuery(PrplMeshBaseTest):
     (for wireless tx links) are not checked in this test.
     """
 
+    @env.process_faults_check
     def runTest(self):
         # skip this test if one of the components does not exist in setup
         try:
diff --git a/tests/boardfarm_plugins/boardfarm_prplmesh/tests/link_metrics.py b/tests/boardfarm_plugins/boardfarm_prplmesh/tests/link_metrics.py
index 86e289448f..785603a7ce 100644
--- a/tests/boardfarm_plugins/boardfarm_prplmesh/tests/link_metrics.py
+++ b/tests/boardfarm_plugins/boardfarm_prplmesh/tests/link_metrics.py
@@ -11,6 +11,7 @@ from capi import tlv
 from opts import debug
 
 import time
+import environment as env
 
 
 class LinkMetrics(PrplMeshBaseTest):
@@ -26,6 +27,7 @@ class LinkMetrics(PrplMeshBaseTest):
         GW - Controller
     '''
 
+    @env.process_faults_check
     def runTest(self):
         # Locate test participants
         try:
diff --git a/tests/boardfarm_plugins/boardfarm_prplmesh/tests/maximum_throughput_tcp.py b/tests/boardfarm_plugins/boardfarm_prplmesh/tests/maximum_throughput_tcp.py
index 49d6eaf5a5..ecaff05f97 100644
--- a/tests/boardfarm_plugins/boardfarm_prplmesh/tests/maximum_throughput_tcp.py
+++ b/tests/boardfarm_plugins/boardfarm_prplmesh/tests/maximum_throughput_tcp.py
@@ -9,11 +9,13 @@ from datetime import datetime
 from opts import debug
 
 import threading
+import environment as env
 
 
 class MaximumThroughputTcp(PrplMeshBaseTest):
     """Check initial configuration on device."""
 
+    @env.process_faults_check
     def runTest(self):
         try:
             agent = self.dev.DUT.agent_entity
diff --git a/tests/boardfarm_plugins/boardfarm_prplmesh/tests/multi_ap_policy_config_w_metric_reporting_policy.py b/tests/boardfarm_plugins/boardfarm_prplmesh/tests/multi_ap_policy_config_w_metric_reporting_policy.py
index 5b70815afb..99c489e987 100644
--- a/tests/boardfarm_plugins/boardfarm_prplmesh/tests/multi_ap_policy_config_w_metric_reporting_policy.py
+++ b/tests/boardfarm_plugins/boardfarm_prplmesh/tests/multi_ap_policy_config_w_metric_reporting_policy.py
@@ -5,6 +5,7 @@
 
 from boardfarm.exceptions import SkipTest
 from common_flow import CommonFlows
+import environment as env
 
 
 class MultiApPolicyConfigWMetricReportingPolicy(CommonFlows):
@@ -15,6 +16,7 @@ class MultiApPolicyConfigWMetricReportingPolicy(CommonFlows):
         GW - Controller
     """
 
+    @env.process_faults_check
     def runTest(self):
         # Locate test participants
         try:
diff --git a/tests/boardfarm_plugins/boardfarm_prplmesh/tests/multi_ap_policy_config_w_steering_policy.py b/tests/boardfarm_plugins/boardfarm_prplmesh/tests/multi_ap_policy_config_w_steering_policy.py
index 7d1e31df6e..87c5d1a178 100644
--- a/tests/boardfarm_plugins/boardfarm_prplmesh/tests/multi_ap_policy_config_w_steering_policy.py
+++ b/tests/boardfarm_plugins/boardfarm_prplmesh/tests/multi_ap_policy_config_w_steering_policy.py
@@ -9,6 +9,7 @@ from capi import tlv
 from opts import debug
 
 import time
+import environment as env
 
 
 class MultiApPolicyConfigWSteeringPolicy(PrplMeshBaseTest):
@@ -18,7 +19,7 @@ class MultiApPolicyConfigWSteeringPolicy(PrplMeshBaseTest):
 
         GW - Controller
     """
-
+    @env.process_faults_check
     def runTest(self):
         # Locate test participants
         try:
diff --git a/tests/boardfarm_plugins/boardfarm_prplmesh/tests/multi_ap_policy_config_w_unsucessful_association.py b/tests/boardfarm_plugins/boardfarm_prplmesh/tests/multi_ap_policy_config_w_unsucessful_association.py
index 431208955e..30eab5972f 100644
--- a/tests/boardfarm_plugins/boardfarm_prplmesh/tests/multi_ap_policy_config_w_unsucessful_association.py
+++ b/tests/boardfarm_plugins/boardfarm_prplmesh/tests/multi_ap_policy_config_w_unsucessful_association.py
@@ -5,6 +5,7 @@
 
 from boardfarm.exceptions import SkipTest
 from common_flow import CommonFlows
+import environment as env
 
 
 class MultiApPolicyConfigWUnsucessfulAssociation(CommonFlows):
@@ -15,7 +16,7 @@ class MultiApPolicyConfigWUnsucessfulAssociation(CommonFlows):
 
         GW - Controller
     """
-
+    @env.process_faults_check
     def runTest(self):
         # Locate test participants
         try:
diff --git a/tests/boardfarm_plugins/boardfarm_prplmesh/tests/nbapi_access_point.py b/tests/boardfarm_plugins/boardfarm_prplmesh/tests/nbapi_access_point.py
index cdf9377338..871f6878cb 100644
--- a/tests/boardfarm_plugins/boardfarm_prplmesh/tests/nbapi_access_point.py
+++ b/tests/boardfarm_plugins/boardfarm_prplmesh/tests/nbapi_access_point.py
@@ -8,13 +8,14 @@ from boardfarm.exceptions import SkipTest
 from opts import debug
 
 import time
+import environment as env
 
 
 class NbapiAccessPoint(PrplMeshBaseTest):
     '''
     This test supposed to test all possible parameters of AccessPoint object..
     '''
-
+    @env.process_faults_check
     def check_bss_is_disabled(self, ssid: str, radio, controller):
         for bss in radio.vaps.values():
             enabled = controller.nbapi_get_parameter(bss.path, "Enabled")
diff --git a/tests/boardfarm_plugins/boardfarm_prplmesh/tests/nbapi_association_event.py b/tests/boardfarm_plugins/boardfarm_prplmesh/tests/nbapi_association_event.py
index 40b7f3ad97..09ad56056e 100644
--- a/tests/boardfarm_plugins/boardfarm_prplmesh/tests/nbapi_association_event.py
+++ b/tests/boardfarm_plugins/boardfarm_prplmesh/tests/nbapi_association_event.py
@@ -14,6 +14,7 @@ from datetime import datetime, timedelta
 import dateutil.parser
 import pytz
 import re
+import environment as env
 
 
 class NbapiAssociationEvent(PrplMeshBaseTest):
@@ -21,7 +22,7 @@ class NbapiAssociationEvent(PrplMeshBaseTest):
     Test for NBAPI Device.WiFi.DataElements.Network.Device.Radio.AssociationEvent object.
     This object describes an event generated when a STA associates to a BSS.
     '''
-
+    @env.process_faults_check
     def runTest(self):
         try:
             controller = self.dev.lan.controller_entity
diff --git a/tests/boardfarm_plugins/boardfarm_prplmesh/tests/nbapi_bss.py b/tests/boardfarm_plugins/boardfarm_prplmesh/tests/nbapi_bss.py
index 13156c1943..bf99b960d4 100644
--- a/tests/boardfarm_plugins/boardfarm_prplmesh/tests/nbapi_bss.py
+++ b/tests/boardfarm_plugins/boardfarm_prplmesh/tests/nbapi_bss.py
@@ -14,6 +14,7 @@ from datetime import datetime, timedelta
 import dateutil.parser
 import pytz
 import re
+import environment as env
 
 
 class NbapiBSS(PrplMeshBaseTest):
@@ -24,6 +25,7 @@ class NbapiBSS(PrplMeshBaseTest):
     def assertEqual(self, name: str, actual: int, expected: str):
         assert actual == int(expected), f"Wrong value for {name}: {actual} expected {expected}"
 
+    @env.process_faults_check
     def runTest(self):
         try:
             agent = self.dev.DUT.agent_entity
diff --git a/tests/boardfarm_plugins/boardfarm_prplmesh/tests/nbapi_cac.py b/tests/boardfarm_plugins/boardfarm_prplmesh/tests/nbapi_cac.py
index ca74c12ac3..2d1a74133d 100644
--- a/tests/boardfarm_plugins/boardfarm_prplmesh/tests/nbapi_cac.py
+++ b/tests/boardfarm_plugins/boardfarm_prplmesh/tests/nbapi_cac.py
@@ -9,6 +9,7 @@ import time
 from .prplmesh_base_test import PrplMeshBaseTest
 from boardfarm.exceptions import SkipTest
 from opts import debug
+import environment as env
 
 
 class NbapiCAC(PrplMeshBaseTest):
@@ -23,7 +24,7 @@ class NbapiCAC(PrplMeshBaseTest):
             CACNonOccupancyChannel.{i}.
             CACActiveChannel.{i}.
     """
-
+    @env.process_faults_check
     def runTest(self):
 
         def check_cac_completion_report(self, agent, channel_pref_report) -> None:
diff --git a/tests/boardfarm_plugins/boardfarm_prplmesh/tests/nbapi_capabilities.py b/tests/boardfarm_plugins/boardfarm_prplmesh/tests/nbapi_capabilities.py
index 4cdf66e8f3..7cf9035cd8 100644
--- a/tests/boardfarm_plugins/boardfarm_prplmesh/tests/nbapi_capabilities.py
+++ b/tests/boardfarm_plugins/boardfarm_prplmesh/tests/nbapi_capabilities.py
@@ -10,6 +10,7 @@ from typing import List
 import sniffer
 
 import time
+import environment as env
 
 
 class NbapiCapabilities(PrplMeshBaseTest):
@@ -50,6 +51,7 @@ class NbapiCapabilities(PrplMeshBaseTest):
                 assert len(non_op_channels) == 1, f"Non-operable channel {channel} was not found."
         supported_op_classes.remove(op_class)
 
+    @env.process_faults_check
     def runTest(self):
         try:
             agent = self.dev.DUT.agent_entity
diff --git a/tests/boardfarm_plugins/boardfarm_prplmesh/tests/nbapi_device.py b/tests/boardfarm_plugins/boardfarm_prplmesh/tests/nbapi_device.py
index 8a9061c6b8..46930e2909 100644
--- a/tests/boardfarm_plugins/boardfarm_prplmesh/tests/nbapi_device.py
+++ b/tests/boardfarm_plugins/boardfarm_prplmesh/tests/nbapi_device.py
@@ -10,13 +10,14 @@ from opts import debug
 from capi import tlv
 
 import time
+import environment as env
 
 
 class NbapiDevice(PrplMeshBaseTest):
     '''
     Test for NBAPI device object.
     '''
-
+    @env.process_faults_check
     def runTest(self):
         try:
             agent = self.dev.DUT.agent_entity
diff --git a/tests/boardfarm_plugins/boardfarm_prplmesh/tests/nbapi_disassociation_event.py b/tests/boardfarm_plugins/boardfarm_prplmesh/tests/nbapi_disassociation_event.py
index 8ba5f2ed4b..fb55fc9d17 100644
--- a/tests/boardfarm_plugins/boardfarm_prplmesh/tests/nbapi_disassociation_event.py
+++ b/tests/boardfarm_plugins/boardfarm_prplmesh/tests/nbapi_disassociation_event.py
@@ -14,6 +14,7 @@ from datetime import datetime, timedelta
 import dateutil.parser
 import pytz
 import re
+import environment as env
 
 
 class NbapiDisassociationEvent(PrplMeshBaseTest):
@@ -22,7 +23,7 @@ class NbapiDisassociationEvent(PrplMeshBaseTest):
     This object describes an event generated when a STA disassociates from a BSS.
     Also, test Network object.
     '''
-
+    @env.process_faults_check
     def runTest(self):
         try:
             controller = self.dev.lan.controller_entity
diff --git a/tests/boardfarm_plugins/boardfarm_prplmesh/tests/nbapi_radio.py b/tests/boardfarm_plugins/boardfarm_prplmesh/tests/nbapi_radio.py
index 6e6cb039db..9220df74f8 100644
--- a/tests/boardfarm_plugins/boardfarm_prplmesh/tests/nbapi_radio.py
+++ b/tests/boardfarm_plugins/boardfarm_prplmesh/tests/nbapi_radio.py
@@ -10,6 +10,7 @@ from capi import tlv
 from opts import debug
 
 import time
+import environment as env
 
 
 class NbapiRadio(PrplMeshBaseTest):
@@ -23,6 +24,7 @@ class NbapiRadio(PrplMeshBaseTest):
         actual = controller.nbapi_get_parameter(path, name)
         assert actual == int(expected), f"Wrong value for {name}: {actual} expected {expected}"
 
+    @env.process_faults_check
     def runTest(self):
         try:
             agent = self.dev.DUT.agent_entity
diff --git a/tests/boardfarm_plugins/boardfarm_prplmesh/tests/nbapi_radio_backhaul_sta.py b/tests/boardfarm_plugins/boardfarm_prplmesh/tests/nbapi_radio_backhaul_sta.py
index dbead41805..b03842f4fb 100644
--- a/tests/boardfarm_plugins/boardfarm_prplmesh/tests/nbapi_radio_backhaul_sta.py
+++ b/tests/boardfarm_plugins/boardfarm_prplmesh/tests/nbapi_radio_backhaul_sta.py
@@ -8,6 +8,7 @@ from .prplmesh_base_test import PrplMeshBaseTest
 from boardfarm.exceptions import SkipTest
 from opts import debug
 import time
+import environment as env
 
 
 class NbapiRadioBackhaulSta(PrplMeshBaseTest):
@@ -16,7 +17,7 @@ class NbapiRadioBackhaulSta(PrplMeshBaseTest):
        Object stores MACAddress of the Backhaul Station (bSTA) on given radio.
        It also checks Backhaul STA Radio Capabilities TLV.
     '''
-
+    @env.process_faults_check
     def runTest(self):
         try:
             agent = self.dev.DUT.agent_entity
diff --git a/tests/boardfarm_plugins/boardfarm_prplmesh/tests/nbapi_scan_result.py b/tests/boardfarm_plugins/boardfarm_prplmesh/tests/nbapi_scan_result.py
index dd6c0089bf..a734ff116d 100644
--- a/tests/boardfarm_plugins/boardfarm_prplmesh/tests/nbapi_scan_result.py
+++ b/tests/boardfarm_plugins/boardfarm_prplmesh/tests/nbapi_scan_result.py
@@ -10,7 +10,7 @@ from capi import tlv
 import sniffer
 
 import time
-
+import environment as env
 # TO DO: Add this test to testsuites after PPM-491 merged.
 # Add only supported channels to Channel Scan Request TLV
 
@@ -102,6 +102,7 @@ class NbapiScanResult(PrplMeshBaseTest):
         pos = ts_len + 12
         self.check_neighbors(neighbor_paths, neighbors_num, pos)
 
+    @env.process_faults_check
     def runTest(self):
         try:
             agent = self.dev.DUT.agent_entity
diff --git a/tests/boardfarm_plugins/boardfarm_prplmesh/tests/nbapi_sta.py b/tests/boardfarm_plugins/boardfarm_prplmesh/tests/nbapi_sta.py
index 88dcb2be1c..0307bfc969 100644
--- a/tests/boardfarm_plugins/boardfarm_prplmesh/tests/nbapi_sta.py
+++ b/tests/boardfarm_plugins/boardfarm_prplmesh/tests/nbapi_sta.py
@@ -14,6 +14,7 @@ from datetime import datetime, timedelta
 import dateutil.parser
 import pytz
 import re
+import environment as env
 
 
 class NbapiSta(PrplMeshBaseTest):
@@ -21,7 +22,7 @@ class NbapiSta(PrplMeshBaseTest):
     This test checks various parameters of NBAPI sta object.
     Path to this object: Device.WiFi.DataElements.Network.Device.Radio.BSS.STA
     '''
-
+    @env.process_faults_check
     def runTest(self):
         try:
             agent = self.dev.DUT.agent_entity
diff --git a/tests/boardfarm_plugins/boardfarm_prplmesh/tests/nbapi_triger_scan.py b/tests/boardfarm_plugins/boardfarm_prplmesh/tests/nbapi_triger_scan.py
index 60c4455a9c..d124374606 100644
--- a/tests/boardfarm_plugins/boardfarm_prplmesh/tests/nbapi_triger_scan.py
+++ b/tests/boardfarm_plugins/boardfarm_prplmesh/tests/nbapi_triger_scan.py
@@ -6,11 +6,12 @@
 from .prplmesh_base_test import PrplMeshBaseTest
 from boardfarm.exceptions import SkipTest
 from opts import debug
+import environment as env
 
 
 class NbapiScanTrigger(PrplMeshBaseTest):
     '''Executes NBAPI ScanTrigger command. '''
-
+    @env.process_faults_check
     def runTest(self):
         try:
             agent = self.dev.DUT.agent_entity
diff --git a/tests/boardfarm_plugins/boardfarm_prplmesh/tests/remote_station.py b/tests/boardfarm_plugins/boardfarm_prplmesh/tests/remote_station.py
index c1365cf25f..06cb3c741b 100644
--- a/tests/boardfarm_plugins/boardfarm_prplmesh/tests/remote_station.py
+++ b/tests/boardfarm_plugins/boardfarm_prplmesh/tests/remote_station.py
@@ -6,11 +6,12 @@
 from .prplmesh_base_test import PrplMeshBaseTest
 from boardfarm.exceptions import SkipTest
 from opts import debug
+import environment as env
 
 
 class RemoteStation(PrplMeshBaseTest):
     """Check initial configuration on device."""
-
+    @env.process_faults_check
     def runTest(self):
         try:
             agent = self.dev.DUT.agent_entity
diff --git a/tests/boardfarm_plugins/boardfarm_prplmesh/tests/resources_test_run.py b/tests/boardfarm_plugins/boardfarm_prplmesh/tests/resources_test_run.py
index a6fb36b5da..bf5e168e19 100644
--- a/tests/boardfarm_plugins/boardfarm_prplmesh/tests/resources_test_run.py
+++ b/tests/boardfarm_plugins/boardfarm_prplmesh/tests/resources_test_run.py
@@ -1,10 +1,11 @@
 from .prplmesh_base_test import PrplMeshBaseTest
 from opts import debug
+import environment as env
 
 
 class ResourcesTestRun(PrplMeshBaseTest):
     """Check initial configuration on device."""
-
+    @env.process_faults_check
     def runTest(self):
 
         agent = self.dev.DUT.agent_entity
diff --git a/tests/boardfarm_plugins/boardfarm_prplmesh/tests/simulate_v2_certification_4_7_10.py b/tests/boardfarm_plugins/boardfarm_prplmesh/tests/simulate_v2_certification_4_7_10.py
index 496fababfa..d878461951 100644
--- a/tests/boardfarm_plugins/boardfarm_prplmesh/tests/simulate_v2_certification_4_7_10.py
+++ b/tests/boardfarm_plugins/boardfarm_prplmesh/tests/simulate_v2_certification_4_7_10.py
@@ -7,6 +7,7 @@ from boardfarm.exceptions import SkipTest
 from capi import tlv
 from common_flow import CommonFlows
 from time import sleep
+import environment as env
 
 
 class V2Certification_4_7_10(CommonFlows):
@@ -18,7 +19,7 @@ class V2Certification_4_7_10(CommonFlows):
         AP1 - Agent1 [DUT]
         GW - Controller
     """
-
+    @env.process_faults_check
     def runTest(self):
         # Locate test participants
         try:
diff --git a/tests/boardfarm_plugins/boardfarm_prplmesh/tests/topology.py b/tests/boardfarm_plugins/boardfarm_prplmesh/tests/topology.py
index 50b43f3619..d06c1ba253 100644
--- a/tests/boardfarm_plugins/boardfarm_prplmesh/tests/topology.py
+++ b/tests/boardfarm_plugins/boardfarm_prplmesh/tests/topology.py
@@ -8,10 +8,11 @@ from boardfarm.exceptions import SkipTest
 from opts import debug
 
 import time
+import environment as env
 
 
 class Topology(PrplMeshBaseTest):
-
+    @env.process_faults_check
     def runTest(self):
         # Locate test participants
         try:
diff --git a/tests/boardfarm_plugins/boardfarm_prplmesh/tests/topology_discovery_burst.py b/tests/boardfarm_plugins/boardfarm_prplmesh/tests/topology_discovery_burst.py
index 9ba637e444..7031b80542 100644
--- a/tests/boardfarm_plugins/boardfarm_prplmesh/tests/topology_discovery_burst.py
+++ b/tests/boardfarm_plugins/boardfarm_prplmesh/tests/topology_discovery_burst.py
@@ -7,6 +7,7 @@ import time
 from boardfarm.exceptions import SkipTest
 from opts import debug
 from .prplmesh_base_test import PrplMeshBaseTest
+import environment as env
 
 
 class TopologyDiscoveryBurst(PrplMeshBaseTest):
@@ -20,7 +21,7 @@ class TopologyDiscoveryBurst(PrplMeshBaseTest):
     source macs is done quickly by the agent. This test imitates step 20 of the R3 4.7.10_ETH test.
 
     """
-
+    @env.process_faults_check
     def runTest(self):
         # Locate test participants
         try:
diff --git a/tests/boardfarm_plugins/boardfarm_prplmesh/tests/tunnelled_frames.py b/tests/boardfarm_plugins/boardfarm_prplmesh/tests/tunnelled_frames.py
index c28baa6012..6f79351419 100644
--- a/tests/boardfarm_plugins/boardfarm_prplmesh/tests/tunnelled_frames.py
+++ b/tests/boardfarm_plugins/boardfarm_prplmesh/tests/tunnelled_frames.py
@@ -7,10 +7,11 @@ from .prplmesh_base_test import PrplMeshBaseTest
 from boardfarm.exceptions import SkipTest
 from opts import debug
 from time import sleep
+import environment as env
 
 
 class TunnelledFrames(PrplMeshBaseTest):
-
+    @env.process_faults_check
     def runTest(self):
         # Locate test participants
         try:
diff --git a/tests/environment.py b/tests/environment.py
index 749b79682c..26d171e880 100644
--- a/tests/environment.py
+++ b/tests/environment.py
@@ -20,6 +20,7 @@ from typing import Dict, Any, List
 
 
 # Third-party imports:
+from boardfarm.exceptions import SkipTest
 import pexpect
 import yaml
 
@@ -416,6 +417,44 @@ def nbapi_ubus_command_not_fail(entity: ALEntity, path: str, command: str,
         return result
 
 
+# Helper function used by the implementations based on ba-cli
+def ba_cli_command(entity: ALEntity, path: str) -> List[str]:
+    command = ["ba-cli", path]
+    result = ""
+    try:
+        debug(f"Executing ba-cli {path}")
+        result = entity.command(*command)
+    except subprocess.CalledProcessError as error:
+        debug(error)
+    return result.split('\n') if result else []
+
+
+# Decorator that adds a prologue and epilogue to a function for checking ProcessFaults;
+# it should only be used with PrplMeshBaseTest class or its subclasses.
+def process_faults_check(func):
+    def wrapper(self, *args, **kwargs):
+        try:
+            DUT = self.dev.DUT.agent_entity
+        except AttributeError as ae:
+            raise SkipTest(ae)
+
+        # We do have support for ProcessFaults only for ALEntityPrplWrt
+        # in other cases we should skip this check
+        if not isinstance(DUT, ALEntityPrplWrt):
+            debug("Unable to perform check because ProcessFaults is not supported")
+            return func(self, *args, **kwargs)
+
+        ba_cli_command(DUT, "ProcessFaults.RemoveAllProcessFaults()")
+        result = func(self, *args, **kwargs)
+        pfaults = ba_cli_command(DUT, "ProcessFaults.ProcessFault.*.?")
+        debug('\n'.join(pfaults))
+
+        assert pfaults[1] == "No data found", "ProcessFaults.ProcessFault. isn't empty"
+
+        return result
+    return wrapper
+
+
 # Concrete implementation with docker
 
 rootdir = os.path.abspath(os.path.join(os.path.dirname(__file__), '..'))
-- 
GitLab

